### PR TITLE
DIG-771: GraphQL-interface Logging Stack Integration - Part 2

### PR DESCRIPTION
**Pull request by AliRZ-02**
## Description
This PR is the second PR associated with the [DIG-771](https://candig.atlassian.net/browse/DIG-771) ticket, aiming to integrate the GraphQL Interface with the CanDIGv2 loggingstack. This PR in particular deals with smaller changes such as submodule updates, and fluentd compatibility updates. The [PR present in the GraphQL-interface repo](https://github.com/CanDIG/GraphQL-interface/pull/12) deals with more of the beefier changes, such as additional print statements, and error traceback formatting. Nevertheless, this PR is required to update the GraphQL-interface submodule with the new logging changes. Fluentd logging has been tested for this PR, though certain changes had to be made such as the version of elasticsearch within the fluentd container, due to version incompatibility. 

## Changelog
- Update GraphQL Submodule within `graphql` subdirectory

## Dependencies
- This PR depends on [Part 1](https://github.com/CanDIG/GraphQL-interface/pull/12) of the CanDIGv2 GraphQL integration ticket, DIG-771

## Testing
- Build-testing this PR is not a requirement due to the minimal changes being made. It is, however, possible to test the logging services with the CanDIGv2 stack.

### Logging Test Setup
- Set up the CanDIGv2 stack as described in the [documentation](https://github.com/CanDIG/CanDIGv2/blob/AliRZ-02/DIG-771-graphql-logging/docs/install-docker.md). Make sure to stop right before the `make images` section, as we will not need to build all of the services for this test. 
- Modify your .env file, changing the `DOCKER_LOG_DRIVER` variable to be equal to `fluentd`. Additionally, modify the `KEYCLOAK_GENERATE_TEST_USER` variable to be `1`. 
- You may need to modify your `lib/logging/fluentd/fluent.conf` file, removing the section nested between the `<match docker.**> and the first </match>` tags. This may be necessary to ensure that the fluent matches are being piped to elasticsearch instead of a logfile. This was the case for me, however, your mileage may vary.
    - eg.  Code Listed Below
```bash
(candig) [alirazazaidi@localhost CanDIGv2]$ git diff lib/logging/fluentd/fluent.conf
diff --git a/lib/logging/fluentd/fluent.conf b/lib/logging/fluentd/fluent.conf
index ff0e917..64986cd 100644
--- a/lib/logging/fluentd/fluent.conf
+++ b/lib/logging/fluentd/fluent.conf
@@ -11,16 +11,6 @@
 </filter>

 <label @mainstream>
-  <match docker.**>
-    @type file
-    @id output_docker1
-    path /fluentd/log/docker.*.log
-    symlink_path /fluentd/log/docker.log
-    append true
-    time_slice_format %Y%m%d
-    time_slice_wait 1m
-    time_format %Y-%m-%dT%H:%M:%S.%N%z
-  </match>
   <match **>
     @type elasticsearch
     @id output_fluentd
@@ -36,4 +26,4 @@
     flush_interval 1s
     template_overwrite true
   </match>
-</label>
\ No newline at end of file
+</label>
```

- Perform the following commands:
```bash
make build-chord-metadata
make build-candig-server
make build-graphql
make build-logging
make compose-logging
make init-authx
make compose-chord-metadata
make compose-candig-server
make compose-graphql
```
- Add testing data if required, as per the previous PR #117 and test per the values listed on the table at the bottom.

### CODE REVIEW

File: lib/logging/fluentd/Dockerfile [hunk 1]
```diff
@@ -6,6 +6,8 @@ LABEL Maintainer="CanDIG Project"
 
 USER root
 
+RUN gem uninstall elasticsearch && gem install elasticsearch -v 7.17.0
```
Comment (shaikh-rashid) [inline lib/logging/fluentd/Dockerfile:9]: Why does elastic version need to be pinned?

**Review by shaikh-rashid — CHANGES_REQUESTED** [review]
LGTM, just need clarification on dependency.

File: lib/logging/fluentd/Dockerfile (hunk 1 above)
Comment (AliRZ-02) [reply lib/logging/fluentd/Dockerfile:9]: A new release (v8.0.0) of elasticsearch about ~3 weeks ago is incompatible with the fluentd-plugin and as such we need to pin the version for now until the release gets fixed. This is simply the solution to the weird logging errors I was mentioning in the candig2-dev channel.

**Review by Zamm178 — APPROVED** [review]
Great work! It seems to be all good from the looks of it

**Review by shaikh-rashid — CHANGES_REQUESTED** [review]
Request for review and possible minor changes

File: etc/env/example.env [hunk 2]
```diff
@@ -323,14 +323,14 @@ CANDIG_DATA_PORTAL_PORT=2543
 CANDIG_DATA_PORTAL_URL=http://${CANDIG_DOMAIN}:${CANDIG_DATA_PORT}/data-portal
 
 # graphql-interface
-GRAPHQL_CONDA_SOURCE=condaforge
-GRAPHQL_CONDA_TYPE=miniforge3
-GRAPHQL_CONDA_VERSION=latest
+GRAPHQL_PYTHON_SOURCE=python
+GRAPHQL_PYTHON_TYPE=3.8
+GRAPHQL_PYTHON_VERSION=slim
```
Comment (shaikh-rashid) [inline etc/env/example.env:328]: Please see comments in https://github.com/CanDIG/GraphQL-interface/pull/11

**Review by shaikh-rashid — APPROVED** [review]
LGTM, just minor change requested but covered in PR https://github.com/CanDIG/GraphQL-interface/pull/11

**Pull request merged**